lib/client/bst-proxy.ts

```diff
@@ -128,16 +129,17 @@ export class BSTProxy {
         return this;
     }
 
-    public start(onStarted?: (error?: any) => void): BSTProxy {
+    public async start(onStarted?: (error?: any) => void): Promise<BSTProxy> {
         // If we have a configuration (i.e., are being run from CLI), we use it
         if (Global.config()) {
             BSTProcess.run(this.httpPort, this.proxyType, process.pid);
             this.proxySecretKey = Global.config().secretKey();
         } else {
             // Handle start when being called programmatically (and presumably standalone)
             if (!this.proxySecretKey) {
-                // If we are being called programmatically, the secret key must be provided
-                throw new Error("Secret key must be provided via .secretKey(key) function. Secret key can be found in ~/.bst/config.");
+                // Load config if not present
+                const config = await BSTConfig.load();
```

Comment:
The code that comes after this I think should (starting from new BespokenClient) I think should go in its own method, say startWithConfig, that gets called once the load is completed.